packages/at_secondary_server/test/sync_unit_test.dart

```diff
@@ -0,0 +1,724 @@
+        var atData = await secondaryPersistenceStore!
+            .getSecondaryKeyStore()!
+            .get('city@alice');
+        expect(atData!.data, 'copenhagen');
+        // Assert the data and metadata stored in the keystore
+        atData = await secondaryPersistenceStore!
+            .getSecondaryKeyStore()!
+            .get('mobile@alice');
+        expect(atData!.data, '1234');
+        expect(atData.metaData!.ttl, 1000);
+        expect(atData.metaData!.ttb, 2000);
+        expect(atData.metaData!.ttr, 3000);
+        expect(atData.metaData!.isCascade, true);
+        // Assert the data and metadata of a public key
+        atData = await secondaryPersistenceStore!
+            .getSecondaryKeyStore()!
+            .get('public:country@alice');
+        expect(atData!.data, 'denmark');
+        expect(atData.metaData!.dataSignature, 'dummy_data_signature');
+        // Assert the key is removed on delete operation
+        expect(
+            secondaryPersistenceStore!
+                .getSecondaryKeyStore()!
+                .isKeyExists('phone@alice'),
+            false);
+      });
+
+      test('test to verify when one of the command in batch has invalid syntax',
+          () async {
+        /// Preconditions
+        /// 1. Server receives the batch request with an invalid command
+        ///
+        /// Assertions
+        /// 1. The valid commands should be processed and commit-id should be added to batch response
+        /// 2. For the invalid batch request command, the error code and error message should be updated in the batch response
+        DefaultVerbHandlerManager().init();
+        var batchRequestCommand = jsonEncode([
+          BatchRequest(1, 'delete:phone@alice'),
+          BatchRequest(2, 'update:city@alice'),
+          BatchRequest(3, 'update:public:country@alice denmark')
+        ]);
+        var batchVerbHandler = BatchVerbHandler(
+            secondaryPersistenceStore!.getSecondaryKeyStore()!);
+        var inBoundSessionId = '_6665436c-29ff-481b-8dc6-129e89199718';
+        var response = Response();
+        var atConnection = InboundConnectionImpl(null, inBoundSessionId);
+        atConnection.metaData.isAuthenticated = true;
+        var batchVerbParams = HashMap<String, String>();
+        batchVerbParams.putIfAbsent('json', () => batchRequestCommand);
+
+        await batchVerbHandler.processVerb(
+            response, batchVerbParams, atConnection);
+        List batchResponseList = jsonDecode(response.data!);
+        expect(batchResponseList.length, 3);
+        expect(batchResponseList[0]['response']['data'], '0');
+        expect(batchResponseList[1]['response']['error_code'], 'AT0003');
+        expect(batchResponseList[1]['response']['error_message'],
+            'Invalid syntax');
+        expect(batchResponseList[2]['response']['data'], '1');
+      });
+      test('Arch - test to verify when invalid json is sent in batch request',
```

Comment:
Does this require an arch discussion? If yes, should we track it through a ticket? I thought we have the server throwing an internal server exception for this case. I mean we do handle invalid JSON already I recon.